src/main/webapp/app/components/calendarButton/UsageAnalysisCalendarButton.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useState } from 'react';
+import { InputGroup, Button, Dropdown, Container } from 'react-bootstrap';
+import classnames from 'classnames';
+import { getMomentInstance } from 'app/shared/utils/Utils';
+import DayPickerInput from 'react-day-picker/DayPickerInput';
+import 'react-day-picker/lib/style.css';
+import moment from 'moment';
+import { APP_LOCAL_DATE_FORMAT } from 'app/config/constants';
+import MomentLocaleUtils from 'react-day-picker/moment';
+import { DefaultTooltip } from 'cbioportal-frontend-commons';
+import pluralize from 'pluralize';
+
+type CalendarButtonProps = {
+  currentDate: string | undefined;
+  currentFromDate: string | undefined;
+  currentToDate: string | undefined;
+  currentToggleState: boolean;
```

Comment:
Shouldn't `currentToggleState` be calculated by currentFromDate and currentToDate?